src/components/QueryEditor.tsx

```diff
@@ -1,59 +1,31 @@
 import React, { PureComponent } from 'react';
 import defaults from 'lodash/defaults';
-import { BracesPlugin, QueryField, Select, Field, Input, Collapse } from '@grafana/ui';
+import { QueryField, Select, Field, Input, Collapse } from '@grafana/ui';
 import { QueryEditorProps, SelectableValue } from '@grafana/data';
-import { DataSource } from './datasource';
-import { defaultQuery, LightstepDataSourceOptions, LightstepQuery } from './types';
+import { DataSource } from '../datasource';
+import { LightstepDataSourceOptions, LightstepQuery } from '../types';
 
 type Props = QueryEditorProps<DataSource, LightstepQuery, LightstepDataSourceOptions>;
-interface QueryEditorState {
-  errorMessage: string;
-  projects: string[];
+type State = {
   isOptionsOpen: boolean;
-}
+};
 
-export class QueryEditor extends PureComponent<Props, QueryEditorState> {
-  plugins: Plugin[] = [BracesPlugin()];
-  state: QueryEditorState = {
+/**
+ * Component responsible for the query text and options editing UI.
+ */
+export class QueryEditor extends PureComponent<Props, State> {
+  state = {
     isOptionsOpen: false,
-    errorMessage: '',
-    projects: [],
   };
 
-  componentDidMount() {
-    this.setState({ projects: this.props.datasource.fetchProjects() });
-  }
+  onQueryChange = (value: string) => {
+    const { onChange, query } = this.props;
 
-  componentWillUnmount() {
-    // Refresh chart if query is deleted
-    if (this.props.onRunQuery) {
-      this.props.onRunQuery();
-    }
-  }
-
-  clearSelections = () => {
-    this.setState({
-      projects: [],
-    });
-  };
-
-  onQueryChange = (value: string, override?: boolean) => {
-    const { onChange, onRunQuery, query } = this.props;
-
-    if (onChange) {
-      onChange({ ...query, text: value });
-
-      if (value === '') {
-        this.clearSelections();
-      }
-
-      if (override && onRunQuery) {
-        onRunQuery();
-      }
```

Comment:
Removed: Override is not part of the function signature for this method's call site and is never passed elsewhere.

<img width="498" alt="image" src="https://user-images.githubusercontent.com/8461733/231759381-3c112c96-cbc7-4949-8a8b-e725cb9501f7.png">